lib/dom-scheduler.js

```diff
@@ -294,7 +294,7 @@
             resolve: resolve
           });
         } else {
-          block();
+          Promise.resolve().then(block);
           resolve();
```

Comment:
I know technically it should dequeued after but it's a bit frightening to read at first.
`Promise.resolve().then(block).then(resolve)` reads a bit better IMO